containers/src/Kokkos_Vector.hpp

```diff
@@ -147,6 +147,72 @@ class vector : public DualView<Scalar*, LayoutLeft, Arg1Type> {
 
   void clear() { _size = 0; }
 
+  void insert(const_iterator it, const value_type& val) { insert(it, 1, val); }
+
+  void insert(const_iterator it, size_type count, const value_type& val) {
+    DV::sync_host();
+    DV::modify_host();
+    if (it < begin() || it >= end())
+      Kokkos::abort("Kokkos::vector::insert : invalid insert iterator");
+
+    ptrdiff_t start = it - begin();
+    auto org_size   = size();
+    resize(size() + count);
```

Comment:
Haven't looked into `resize()` implementation but I would return early if `count == 0`.

Reply:
resize does that.